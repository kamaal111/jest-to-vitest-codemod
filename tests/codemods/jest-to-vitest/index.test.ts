import { describe, expect, it } from 'vitest';

import { invalidRuleSignal } from '../../test-utils/detection-theory';
import {
  JEST_TO_VITEST_LANGUAGE,
  jestToVitestModifications,
  makeJestToVitestInitialModification,
} from '../../../src/codemods/jest-to-vitest';

describe('jest.SpyInstance -> MockInstance', () => {
  it('replaces jest SpyInstance with vi MockInstance', async () => {
    const source = `
    type Spied = jest.SpyInstance<string>
    
    beforeEach(() => { setActivePinia(createTestingPinia()) })
    `;
    const modifications = await invalidRuleSignal(source, JEST_TO_VITEST_LANGUAGE, ast => {
      return jestToVitestModifications(makeJestToVitestInitialModification(ast));
    });
    const updatedSource = modifications.ast.root().text();

    expect(updatedSource).not.toContain(`SpyInstance`);
    expect(updatedSource).toContain(`MockInstance<string>`);
    expect(updatedSource).toContain('type MockInstance');
  });
});
