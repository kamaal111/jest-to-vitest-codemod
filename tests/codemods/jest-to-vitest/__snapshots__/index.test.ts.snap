// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`jest.SpyInstance -> MockInstance > replaces jest SpyInstance with vi MockInstance 1`] = `
"import { beforeEach, type MockInstance } from 'vitest';

type Spied = MockInstance<string>
    
    beforeEach(() => { setActivePinia(createTestingPinia()) })"
`;
